src/main/scala/cromwell/engine/workflow/WorkflowActor.scala

```diff
@@ -552,6 +558,12 @@ case class WorkflowActor(workflow: WorkflowDescriptor, backend: Backend)
     context.system.scheduler.scheduleOnce(ResendInterval, self, message)
   }
 
+  private def removePendingCallKeyPersistence(data: WorkflowData, message: PersistenceMessage) = {
+    logger.debug(s"Got whenUnhandled message: $message")
+    val updatedData = data.removePersisting(message.callKey, message.executionStatus)
+    stay using updatedData
```

Comment:
Hmm, I'm not sure I really like this return type either. But now I see this is very specifically a message handling method so maybe neither are quite so terrible a I first though